testing/prow/gpu-operator.sh

```diff
@@ -109,30 +109,57 @@ prepare_cluster_for_gpu_operator() {
 
 collect_must_gather() {
     run_in_sub_shell() {
-        echo "Running gpu-operator_gather ..."
-        /usr/bin/gpu-operator_gather &> /dev/null
-
-        # extract ARTIFACT_EXTRA_LOGS_DIR from 'source toolbox/_common.sh' without sourcing it directly
-
-        export TOOLBOX_SCRIPT_NAME=toolbox/gpu-operator/must-gather.sh
-        COMMON_SH=$(
-            bash -c 'source toolbox/_common.sh;
-                     echo "8<--8<--8<--";
-                     # only evaluate these variables from _common.sh
-                     env | egrep "(^ARTIFACT_EXTRA_LOGS_DIR=)"'
-                 )
-        ENV=$(echo "$COMMON_SH" | tac | sed '/8<--8<--8<--/Q' | tac) # keep only what's after the 8<--
-        eval $ENV
+        echo "Running the GPU Operator must-gather image ..."
+        OPERATOR_IMAGE=$(oc get pods -A -lapp=gpu-operator -o=jsonpath='{range .items[*]}{range .spec.containers[*]}{.image}{end}{end}' || true)
```

Comment:
Is the operator pod the only pod that has label `app=gpu-operator`?

This will break (more than one result) if:
- other pods get this label 
- new containers are added to the operator pod
- the operator pod gets deleted, in that case there will temporarily be 2 operator pods

not sure how likely are any of those, but consider making a more robust query (e.g. find a better label if there is one, filter the list of containers by name rather than taking all of them, look only at the first pod, etc)

Reply:
good point,
I think in this CI entrypoint script, it's fair to assume that no other Pod will match `app=gpu-operator`

I've updated the `jsonpath` to return only one image if there are multiple Pods/Containers:

```
oc get pods -A -lapp=gpu-operator -o=jsonpath='{.items[].spec.containers[].image}'
```

Reply:
Maybe change [] and [] to [0] and [0] just to clarify intent